src/parser.cc

```diff
@@ -685,6 +685,17 @@ bool readDoc(tinyxml2::XMLDocument *_xmlDoc, SDFPtr _sdf,
           "Error reading element <" + _sdf->Root()->GetName() + ">"});
       return false;
     }
+
+    // delimiter '::' in element names not allowed in SDFormat >= 1.8
+    if (std::stof(_sdf->Root()->OriginalVersion()) >= 1.8f
```

Comment:
this logic won't work when we reach specification version `1.10`. I recommend using `ignition::math::SemanticVersion`

* https://github.com/osrf/sdformat/blob/sdformat9_9.5.0/src/Model.cc#L161
* https://github.com/osrf/sdformat/blob/sdformat9_9.5.0/src/Model.cc#L247

Reply:
Thanks! 56e47a73303264cfeab1c7ee4aebd14083cee0dd